### PR TITLE
fix(jinja): handle non-JSON body with JSON content-type in get_form_data

### PR DESCRIPTION
**Pull request by aminghadersohi**
### SUMMARY
`get_form_data()` crashed with an unhandled `werkzeug.exceptions.BadRequest` whenever it ran inside a request context where the `Content-Type` header claimed JSON but the body wasn't actually parseable JSON. This happens for any caller that reaches `get_form_data()` outside a normal chart-data POST — for example the `filter_values()` / `get_filters()` Jinja macros used in chart and dataset SQL templates, when rendered from a request context that isn't a real chart-data request.

`request.is_json` only inspects the `Content-Type` header; it does not guarantee the body is valid, non-empty JSON. `get_form_data()` already has a helper, `loads_request_json()`, that gracefully falls back to `{}` on a JSON parse failure — but that guard was only applied to the `request.form` / `request.args` fallback paths, not to the primary `request.get_json(cache=True)` call, which is the one that was actually raising.

This PR extracts the JSON-body parsing into a small `get_request_json_body()` helper that catches `BadRequest` and falls back to `{}`, matching the existing `loads_request_json()` behavior.

Internal tracking: SC-114072

### TESTING INSTRUCTIONS
- Added `tests/unit_tests/views/test_utils.py::test_get_form_data_handles_non_json_body_with_json_content_type`, which builds a request context with `Content-Type: application/json` and a non-JSON body, and asserts `get_form_data()` returns `({}, None)` instead of raising.
- `pytest tests/unit_tests/views/test_utils.py tests/unit_tests/jinja_context_test.py -q` — 104 passed.
- Manually reproduced the crash against `master` (`request.get_json(cache=True)` raising `BadRequest`) and confirmed the fix resolves it via `app.test_request_context(data="not-json-at-all", content_type="application/json")`.

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/views/utils.py [hunk 1]
```diff
@@ -214,6 +215,24 @@ def loads_request_json(request_json_data: str) -> dict[Any, Any]:
...
+    ``filter_values()``) can have a request context whose Content-Type claims
+    JSON but whose body isn't a JSON chart-data payload, which makes Werkzeug
+    raise ``BadRequest`` from ``request.get_json()``.
+    """
+    if not request.is_json:
+        return {}
+    try:
+        return request.get_json(cache=True)
+    except BadRequest:
+        return {}
```
[the remark this answers is not in the document]
Comment (aminghadersohi) [reply superset/views/utils.py:218]: Confirmed — good catch. `request.get_json(cache=True)` can return a well-formed but non-dict JSON value (`null`, a scalar, or an array), which `get_form_data()` then treats as a mapping. Fixed in the latest push: `get_request_json_body()` now returns `{}` unless the parsed body is actually a `dict`, and I added a regression test (`test_get_form_data_handles_non_dict_json_body`) covering a scalar JSON body. Also rebased onto current `master` to resolve the merge conflict from #42126 removing `JS_CONTROL_FORM_DATA_KEYS`. See commit 7a1083e.

**Comment by aminghadersohi** [comment]
Rebased onto current `master` (resolves the conflict from #42126, which removed `JS_CONTROL_FORM_DATA_KEYS`/`REJECTED_FORM_DATA_KEYS` from the same file/test module) and addressed the [non-dict JSON body finding](https://github.com/apache/superset/pull/42196#discussion_r3627315762) — `get_request_json_body()` now coerces any non-dict JSON body (`null`, scalar, array) to `{}`, with a new regression test. PR is up to date and mergeable.

**Review by rusackas — APPROVED** [review]
Thanks for tracking this down, and for circling back to coerce the non-dict JSON case too once Bito flagged it. Regression tests cover both edge cases and CI's green. LGTM, will merge once nobody else objects.
